lib/dependency.js

```diff
@@ -0,0 +1,248 @@
+'use strict';
+
+var child_process = require('child_process');
+var fs = require('./file.js');
+var path = require('path');
+var req = require('lazreq')({
+  dapphub: './dapphub_registry.js',
+  deasync: 'deasync',
+  ipfsAPI: 'ipfs-api',
+  os: 'os',
+  Web3Factory: './web3Factory.js',
+  Workspace: './workspace.js'
+});
+var semver = require('semver');
+
+module.exports = class Dependency {
+  constructor (path, version, name) {
+    this.setName(name || '');
+    this.path = path || '';
+    this.version = version || '';
+    this.packagesDirectory = 'dapple_packages';
```

Comment:
Code smell: Probably should be pulling this from some global constants object instead.